core/templates/pages/exploration-player-page/learner-experience/tutor-card.component.ts

```diff
@@ -139,6 +139,12 @@ export class TutorCardComponent {
     this.directiveSubscriptions.unsubscribe();
   }
 
+  ngOnChanges(changes: SimpleChanges): void {
```

Comment:
This change was required since currently in develop, I faced an issue where the interaction instructions weren't visible, and it was due to past state data persisting in this component at the time of this service updation. This fixed it.

Reply:
How frequently does this get called? Depending on that we can decide whether to explicilty check if there are changes to properties within the SimpleChanges object before executing the rest of the function.

Reply:
From my quick testing, this was called once, every time the card was changed in the learner view.
So, `changes.displayedCard` is present only if the `displayedCard` object has a change. So, this only calls the function if the card object has changed currently. Do you mean a deeper check into the fields themselves?

Reply:
Ah I see, should we do something like this:

```ts
let displayedCard = changes.displayedCard;
if (displayedCard && displayedCard.currentValue !== displayedCard.previousValue) {
  this.updateDisplayedCard();
}
```

So that we can be sure that this executes only when the displayedCard changes and not any other field?

Reply:
Done!